src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test.Common/CSharpTestLspServerHelpers.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.ExternalAccess.Razor;
+using Microsoft.CodeAnalysis.Razor.Workspaces.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Composition;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.Common
+{
+    internal class CSharpTestLspServerHelpers
+    {
+        public static async Task<TResponse> ExecuteCSharpRequestAsync<TRequest, TResponse>(
+            RazorCodeDocument codeDocument,
+            Uri csharpDocumentUri,
+            ServerCapabilities serverCapabilities,
+            TRequest requestParams,
+            string methodName,
+            CancellationToken cancellationToken) where TRequest : class
+        {
+            var csharpSourceText = codeDocument.GetCSharpSourceText();
+            var files = new List<(Uri, SourceText)>
+            {
+                (csharpDocumentUri, csharpSourceText)
+            };
+
+            var exportProvider = RoslynTestCompositions.Roslyn.ExportProviderFactory.CreateExportProvider();
+            using var workspace = CreateCSharpTestWorkspace(files, exportProvider);
+            await using var csharpLspServer = await CreateCSharpLspServerAsync(workspace, exportProvider, serverCapabilities);
+
+            var result = await csharpLspServer.ExecuteRequestAsync<TRequest, TResponse>(
+                methodName,
+                requestParams,
+                cancellationToken).ConfigureAwait(false);
+
+            return result;
+        }
+
+        private static AdhocWorkspace CreateCSharpTestWorkspace(IEnumerable<(Uri documentUri, SourceText csharpSourceText)> files, ExportProvider exportProvider)
```

Comment:
Food for thought: Might be nice to create a `private record struct` instead of a named tuple, the named tuple pattern leaks like crazy so it's typically good to capture the representation

I know this method was copied over so don't feel obligated to address the feedback, noticed it so figured I'd leave a comment 😄